deny.toml

```diff
@@ -0,0 +1,49 @@
+targets = []
+
+[advisories]
+db-path = "~/.cargo/advisory-db"
+db-urls = ["https://github.com/rustsec/advisory-db"]
+vulnerability = "deny"
+unmaintained = "warn"
+yanked = "warn"
+notice = "warn"
+ignore = []
+
+[licenses]
+unlicensed = "deny"
+allow = [
+    "Apache-2.0",
+    "Apache-2.0 WITH LLVM-exception",
+    "BSD-3-Clause",
+    "ISC",
+    "MIT",
+]
+deny = []
+copyleft = "deny"
+allow-osi-fsf-free = "neither"
+default = "deny"
+confidence-threshold = 0.8
+exceptions = []
+
+[bans]
+multiple-versions = "warn"
+# Lint level for when a crate version requirement is `*`
+wildcards = "deny"
+highlight = "all"
+allow = []
+deny = [
+    { name = "ring" },
+]
+skip = []
+skip-tree = []
```

Comment:
:thought_balloon: Probably not needed with our dependabot reviews, but there is [`allow-build-scripts`](https://embarkstudios.github.io/cargo-deny/checks/bans/cfg.html#the-allow-build-scripts-field-optional), which can avoid crates adding a build script that isn't looked at.

Reply:
Hmm, interesting. I'm a bit torn. I think it's an abstract good that we have documented which dependencies have build scripts (so we can allowlist them), but as you say, there may not be a practical benefit to doing so...

I think we should leave it out for now, based on thorough dependabot reviews.